src/HttpHeader.cc

```diff
@@ -736,6 +736,37 @@ HttpHeader::refreshMask()
     }
 }
 
+bool
+HttpHeader::tooLarge() const
+{
+    size_t max = 0;
+    switch (owner) {
+    case hoRequest:
+        max = Config.maxRequestHeaderSize;
+        break;
+    case hoReply:
+        max = Config.maxReplyHeaderSize;
+        break;
+#if USE_HTCP
+    case hoHtcpReply:
+        // HTCP messages have a fixed header length
+        return false;
+#endif
+#if USE_OPENSSL
+    case hoErrorDetail:
+        // this case should not happen
+        return false;
+#endif
+    case hoNone:
+        // this case should not happen
```

Comment:
Same "why" question/concern here. hoNone is used in the default HttpHeader constructor, so we cannot easily claim that there are no HttpHeader objects with hoNone owners somewhere. And if such objects do not currently exist, then we should remove that constructor itself as well AFAICT.

Reply:
> should remove that constructor itself as well

I removed this (unused) constructor. However I did not remove `hoNone` because it is used (indirectly) in `HttpHeaderStats`.